### PR TITLE
Add Menu component

### PR DESCRIPTION
**Pull request by tamorim**
Implements Menu component!
Depends on https://github.com/vtex-apps/storefront-sdk/pull/16 and #97

### CODE REVIEW

**Comment by tamorim** [comment]
@brenoc

File: src/components/Header/Header.js [hunk 1]
```diff
@@ -7,10 +9,27 @@ import searchIcon from 'assets/icons/search.svg';
...
+  state = {
+    isMenuOpen: false
+  }
+
+  handleMenuTap = () => {
+    this.setState({ isMenuOpen: !this.state.isMenuOpen});
+  }
+
   render() {
+    let menu = this.state.isMenuOpen ?
```
Comment (brenoc) [inline src/components/Header/Header.js]: What do you think of using a function to do this instead?

Something like this:
```js
renderMenuElement = () => {
  if (!this.state.isMenuOpen) return null;
  
  return (
      <Area key="Menu"
            id={`${this.props.areaPath}/menu`}
            toggleMenu={this.handleMenuTap}/>
    );
  }
}

render() {
  const menu = this.renderMenuElement();
  ...
}
```

Comment (tamorim) [inline src/components/Header/Header.js]: @brenoc I actually prefer like this >:

**Comment by tamorim** [comment]
@brenoc Updated to NavigationMenu!
Will publish NavigationMenu later :)

**Comment by tamorim** [comment]
@brenoc Can you review this, please? :smiley:

**Comment by brenoc** [comment]
@tamorim I just merged a pull request from @dfosco is this branch rebased?

**Comment by tamorim** [comment]
@brenoc Yup!

**Comment by tamorim** [comment]
@brenoc Wait, I think I didn't push some updates

**Pull request merged**